plugins/pulp_rpm/plugins/importers/yum/associate.py

```diff
@@ -253,8 +254,9 @@ def copy_rpms(units, source_repo, dest_repo, import_conduit, config, solver=None
     if solver:
         # This returns units that have a flattened 'provides' metadata field
         # for memory purposes (RHBZ #1185868)
-        deps = solver.find_dependent_rpms(unit_set)
-        unit_set |= deps
+        repo_unit_set = solver.find_dependent_rpms(unit_set)
```

Comment:
I think `repo_unit_set` is not used for identifying which unit to copy to which target repo and currently it copies everything into one. Is it intentional? Do you think it's out of the scope of this PR?
If yes, then we should make it clear on the issue that this is one of the items which remains to be implemented.

Reply:
> Is it intentional? Do you think it's out of the scope of this PR?

Yes x2.  I'll leave a comment.